File: BlackFireIM/Example/ChatDetail/Controller/BFWinkMessageCell..m
//
//  MSWinkMessageCell.m
//  BlackFireIM
//
//  Created by benny wang on 2021/4/13.
//

#import "BFWinkMessageCell.h"
#import "BFWinkMessageCellData.h"
#import "MSHeader.h"


@implementation BFWinkMessageCell

- (instancetype)initWithStyle:(UITableViewCellStyle)style reuseIdentifier:(NSString *)reuseIdentifier
{
    self = [super initWithStyle:style reuseIdentifier:reuseIdentifier];
    if (self) {
        _animationView = [LOTAnimationView animationNamed:@"wink"];
        _animationView.loopAnimation = YES;
        [self.container addSubview:_animationView];
        _animationView.frame = self.container.bounds;
        _animationView.autoresizingMask = UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight;
        [_animationView play];
        
        _noticeL = [[UILabel alloc]init];
        _noticeL.font = [UIFont systemFontOfSize:12];
        _noticeL.text = @"like";
        _noticeL.textAlignment = NSTextAlignmentRight;
        _noticeL.textColor = [UIColor d_systemGrayColor];
        [self.container addSubview:_noticeL];
    }
    return self;
}

- (void)fillWithData:(BFWinkMessageCellData *)data
{
    //set data
    [super fillWithData:data];
}

- (void)layoutSubviews
{
    [super layoutSubviews];
    self.noticeL.frame = CGRectMake(self.container.width-50, self.container.height-15, 50, 15);
}

@end

File: Swift/MSIM-Swift/ChatDetail/C2C/BFWinkMessageCell.swift
//
//  BFWinkMessageCell.swift
//  MSIM-Swift
//
//  Created by benny wang on 2021/9/1.
//

import UIKit
import Lottie


class BFWinkMessageCell: MSMessageCell {

    var animationView: AnimationView!
    
    var noticeL: UILabel!
    
    override init(style: UITableViewCell.CellStyle, reuseIdentifier: String?) {
        super.init(style: style, reuseIdentifier: reuseIdentifier)
        
        animationView = AnimationView(name: "wink")
        animationView.loopMode = .loop
        container.addSubview(animationView)
        animationView.frame = container.bounds
        animationView.autoresizingMask = [.flexibleWidth,.flexibleHeight]
        animationView.play()
        
        noticeL = UILabel()
        noticeL.font = .systemFont(ofSize: 12)
        noticeL.text = "like"
        noticeL.textAlignment = .right
        noticeL.textColor = UIColor.d_color(light: MSMcros.TText_Color, dark: MSMcros.TText_Color_Dark)
        container.addSubview(noticeL)
    }
    
    override func fillWithData(data: MSMessageCellData) {
        super.fillWithData(data: data)
        
    }
    
    override func layoutSubviews() {
        super.layoutSubviews()
        noticeL.frame = CGRect(x: container.width - 50, y: container.height - 15, width: 50, height: 15)
    }
    
    required public init?(coder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
    
}
